# Exploratory Data Analysis (EDA) of the Steam Store Games dataset
 ## Data consists of steam store games and their attributes
 ### Dataset was obtained from Kaggle (https://www.kaggle.com/nikdavis/steam-store-games)

#### Dataset includes over 27,000 games, the primary goal is to explore the dataset and identify any trends or patterns that may be present in the data . By doing so, we can gain a better understanding of the dataset and potentially identify any interesting insights which can be used for further analysis.

## PART 1 steam.csv - Dataset containing information about the games

## Importing necessary libraries

In [25]:
import pandas as pd #Data manipulation and analysis
import numpy as np #Numerical computing
import matplotlib.pyplot as plt #Data visualization
import seaborn as sns #Data visualization on top of matplotlib

### Loading the dataset

In [26]:
df = pd.read_csv('Data/SteamData/steam.csv')
df.head()

appid                       name release_date  english         developer  \
0     10             Counter-Strike   2000-11-01        1             Valve   
1     20      Team Fortress Classic   1999-04-01        1             Valve   
2     30              Day of Defeat   2003-05-01        1             Valve   
3     40         Deathmatch Classic   2001-06-01        1             Valve   
4     50  Half-Life: Opposing Force   1999-11-01        1  Gearbox Software   

  publisher          platforms  required_age  \
0     Valve  windows;mac;linux             0   
1     Valve  windows;mac;linux             0   
2     Valve  windows;mac;linux             0   
3     Valve  windows;mac;linux             0   
4     Valve  windows;mac;linux             0   

                                          categories  genres  \
0  Multi-player;Online Multi-Player;Local Multi-P...  Action   
1  Multi-player;Online Multi-Player;Local Multi-P...  Action   
2              Multi-player;Valve Anti-Cheat enabled  Action   
3  Multi-player;Online Multi-Player;Local Multi-P...  Action   
4  Single-player;Multi-player;Valve Anti-Cheat en...  Action   

                  steamspy_tags  achievements  positive_ratings  \
0        Action;FPS;Multiplayer             0            124534   
1        Action;FPS;Multiplayer             0              3318   
2  FPS;World War II;Multiplayer             0              3416   
3        Action;FPS;Multiplayer             0              1273   
4             FPS;Action;Sci-fi             0              5250   

   negative_ratings  average_playtime  median_playtime             owners  \
0              3339             17612              317  10000000-20000000   
1               633               277               62   5000000-10000000   
2               398               187               34   5000000-10000000   
3               267               258              184   5000000-10000000   
4               288               624              415   5000000-10000000   

   price  
0   7.19  
1   3.99  
2   3.99  
3   3.99  
4   3.99

In [27]:
df.shape

(27075, 18)

As we can see , the dataset contains 27,075 rows and 18 columns , which is a good amount of data to work with.

In [28]:
df.dtypes # Checking the data types of the columns

appid                 int64
name                 object
release_date         object
english               int64
developer            object
publisher            object
platforms            object
required_age          int64
categories           object
genres               object
steamspy_tags        object
achievements          int64
positive_ratings      int64
negative_ratings      int64
average_playtime      int64
median_playtime       int64
owners               object
price               float64
dtype: object

In [29]:
df.isnull().sum()

appid                0
name                 0
release_date         0
english              0
developer            1
publisher           14
platforms            0
required_age         0
categories           0
genres               0
steamspy_tags        0
achievements         0
positive_ratings     0
negative_ratings     0
average_playtime     0
median_playtime      0
owners               0
price                0
dtype: int64

Further we can see that the dataset contains some missing values in collumns developer and publusher, which we will deal with later, but for the further analysis we need to dislay the exact null values.

In [30]:
# checking the exact null values

df[df['developer'].isnull()]

appid                name release_date  english developer publisher  \
23071  881910  The Battle Of Ages   2018-09-07        1       NaN       NaN   

      platforms  required_age                        categories  \
23071   windows             0  Multi-player;Online Multi-Player   

                                                  genres  \
23071  Action;Indie;Massively Multiplayer;RPG;Early A...   

                                   steamspy_tags  achievements  \
23071  Early Access;Action;Massively Multiplayer             0   

       positive_ratings  negative_ratings  average_playtime  median_playtime  \
23071                 2                 1                 0                0   

        owners  price  
23071  0-20000   3.99

In [31]:

df[df['publisher'].isnull()]

appid                                   name release_date  english  \
3420   307170                               Borealis   2014-09-02        1   
4511   341120  Glorkian Warrior: The Trials Of Glork   2015-03-24        1   
5229   359370                          Pirate's Life   2015-04-17        1   
7464   422940                     Divergence: Online   2016-01-06        1   
7911   436240                    Melancholy Republic   2018-04-12        1   
9894   498710                           After Dreams   2018-05-06        1   
10011  502150             Interstellar Logistics Inc   2016-08-15        1   
10564  516430                   Ruin of the Reckless   2017-04-26        1   
11123  531240                              Max Stern   2016-10-21        1   
13901  610740                     SuperCluster: Void   2017-05-15        1   
17829  727280                                  Scrap   2017-10-18        1   
22867  875710                      BRANE™: Prototype   2018-07-01        1   
23071  881910                     The Battle Of Ages   2018-09-07        1   
23876  911050                         Holiday Escape   2018-08-18        1   

                                               developer publisher  \
3420                                       Conrad Nelson       NaN   
4511                                            Pixeljam       NaN   
5229                                       Team Eyepatch       NaN   
7464                                 Stained Glass Llama       NaN   
7911                                Cloud Runner Studios       NaN   
9894                                          Matt Boyer       NaN   
10011                            Exalted Guy Interactive       NaN   
10564                               Faux-Operative Games       NaN   
11123                                  Lupan Artiom Oleg       NaN   
13901                                      Logan McClure       NaN   
17829                                  Top Shelf Studios       NaN   
22867                                         John Wanot       NaN   
23071                                                NaN       NaN   
23876  Kit Gardner;Sam Robinson;Holly Gardner;Jason R...       NaN   

               platforms  required_age  \
3420             windows             0   
4511         windows;mac             0   
5229             windows             0   
7464             windows             0   
7911             windows             0   
9894             windows             0   
10011            windows             0   
10564            windows             0   
11123  windows;mac;linux             0   
13901            windows             0   
17829            windows             0   
22867      windows;linux             0   
23071            windows             0   
23876            windows             0   

                                              categories  \
3420   Single-player;Steam Achievements;Steam Trading...   
4511   Single-player;Steam Achievements;Full controll...   
5229                                       Single-player   
7464           Multi-player;MMO;Steam Turn Notifications   
7911                                       Single-player   
9894                                       Single-player   
10011                                      Single-player   
10564  Single-player;Co-op;Local Co-op;Shared/Split S...   
11123  Single-player;Steam Achievements;Steam Trading...   
13901                                      Single-player   
17829  Single-player;Local Multi-Player;Local Co-op;S...   
22867                                      Single-player   
23071                   Multi-player;Online Multi-Player   
23876                                      Single-player   

                                                  genres  \
3420                                 Action;Casual;Indie   
4511                                               Indie   
5229                                 Simulation;Strategy   
7464   Action;

## Data Cleaning
We will start by cleaning the data, which includes handling missing values, removing duplicates, and dropping unnecessary columns.

In [32]:
# Dropping the missing values
df.dropna(subset=['developer','publisher'],inplace=True)

In [33]:
df.isnull().sum() # Checking if the missing values have been removed

appid               0
name                0
release_date        0
english             0
developer           0
publisher           0
platforms           0
required_age        0
categories          0
genres              0
steamspy_tags       0
achievements        0
positive_ratings    0
negative_ratings    0
average_playtime    0
median_playtime     0
owners              0
price               0
dtype: int64

In [34]:
df.nunique() # Checking the unique values in each column

appid               27061
name                27019
release_date         2619
english                 2
developer           17102
publisher           14353
platforms               7
required_age            6
categories           3331
genres               1551
steamspy_tags        6420
achievements          410
positive_ratings     2800
negative_ratings     1492
average_playtime     1345
median_playtime      1312
owners                 13
price                 282
dtype: int64

We can see that most of the columns have a large number of unique values, which is expected as the dataset contains information about over 27,000 games , but some of the columns have the similar values , such as required_age and platforms.


In [35]:
df.describe() # Statistical information about the dataset
# Takes only numerical columns into consideration

appid       english  required_age  achievements  \
count  2.706100e+04  27061.000000  27061.000000  27061.000000   
mean   5.962192e+05      0.981117      0.355087     45.270278   
std    2.509174e+05      0.136115      2.406653    352.760212   
min    1.000000e+01      0.000000      0.000000      0.000000   
25%    4.012100e+05      1.000000      0.000000      0.000000   
50%    5.990900e+05      1.000000      0.000000      7.000000   
75%    7.987900e+05      1.000000      0.000000     23.000000   
max    1.069460e+06      1.000000     18.000000   9821.000000   

       positive_ratings  negative_ratings  average_playtime  median_playtime  \
count      2.706100e+04      27061.000000      27061.000000     27061.000000   
mean       1.001042e+03        211.120579        149.847197       146.095821   
std        1.899362e+04       4286.044728       1827.507970      2354.486857   
min        0.000000e+00          0.000000          0.000000         0.000000   
25%        6.000000e+00          2.000000          0.000000         0.000000   
50%        2.400000e+01          9.000000          0.000000         0.000000   
75%        1.260000e+02         42.000000          0.000000         0.000000   
max        2.644404e+06     487076.000000     190625.000000    190625.000000   

              price  
count  27061.000000  
mean       6.079350  
std        7.876308  
min        0.000000  
25%        1.690000  
50%        3.990000  
75%        7.190000  
max      421.990000

In [36]:
df['required_age'].value_counts() # Value counts of the required age

required_age
0     26465
18      308
16      192
12       73
7        12
3        11
Name: count, dtype: int64

In [37]:
import duckdb

In [38]:
%%sql
SELECT required_age FROM df GROUP BY required_age

required_age
0             7
1            18
2            12
3             3
4             0
5            16

In [40]:
df['platforms'] = df['platforms'].str.split(';') # Splitting the platforms column

In [44]:
df

appid                        name release_date  english  \
0           10              Counter-Strike   2000-11-01        1   
1           20       Team Fortress Classic   1999-04-01        1   
2           30               Day of Defeat   2003-05-01        1   
3           40          Deathmatch Classic   2001-06-01        1   
4           50   Half-Life: Opposing Force   1999-11-01        1   
...        ...                         ...          ...      ...   
27070  1065230             Room of Pandora   2019-04-24        1   
27071  1065570                   Cyber Gun   2019-04-23        1   
27072  1065650            Super Star Blast   2019-04-24        1   
27073  1066700  New Yankee 7: Deer Hunters   2019-04-17        1   
27074  1069460                   Rune Lord   2019-04-24        1   

                developer             publisher              platforms  \
0                   Valve                 Valve  [windows, mac, linux]   
1                   Valve                 Valve  [windows, mac, linux]   
2                   Valve                 Valve  [windows, mac, linux]   
3                   Valve                 Valve  [windows, mac, linux]   
4        Gearbox Software                 Valve  [windows, mac, linux]   
...                   ...                   ...                    ...   
27070         SHEN JIAWEI           SHEN JIAWEI              [windows]   
27071      Semyon Maximov      BekkerDev Studio              [windows]   
27072         EntwicklerX           EntwicklerX              [windows]   
27073  Yustas Game Studio  Alawar Entertainment         [windows, mac]   
27074    Adept Studios GD  Alawar Entertainment         [windows, mac]   

       required_age                                         categories  \
0                 0  Multi-player;Online Multi-Player;Local Multi-P...   
1                 0  Multi-player;Online Multi-Player;Local Multi-P...   
2                 0              Multi-player;Valve Anti-Cheat enabled   
3                 0  Multi-player;Online Multi-Player;Local Multi-P...   
4                 0  Single-player;Multi-player;Valve Anti-Cheat en...   
...             ...                                                ...   
27070             0                   Single-player;Steam Achievements   
27071             0                                      Single-player   
27072             0  Single-player;Multi-player;Co-op;Shared/Split ...   
27073             0                          Single-player;Steam Cloud   
27074             0                          Single-player;Steam Cloud   

                       genres                 steamspy_tags  achievements  \
0                      Action        Action;FPS;Multiplayer             0   
1                      Action        Action;FPS;Multiplayer             0   
2                      Action  FPS;World War II;Multiplayer             0   
3                      Action        Action;FPS;Multiplayer             0   
4                      Action             FPS;Action;Sci-fi             0   
...                       ...                           ...           ...   
27070  Adventure;Casual;Indie        Adventure;Indie;Casual             7   
27071  Action;Adventure;Indie        Action;Indie;Adventure             0   
27072     Action;Casual;Indie           Action;Indie;Casual            24   
27073  Adventure;Casual;Indie        Indie;Casual;Adventure             0   
27074  Adventure;Casual;Indie        Indie;Casual;Adventure             0   

       positive_ratings  negative_ratings  average_playtime  median_playtime  \
0                124534              3339             17612              317   
1                  3318               633               277               62   
2                  3416               398               187               34   
3                  1273               267               258              184   
4                  5250               288               624              415   
...      

In [46]:
platforms = df['platforms'].explode().value_counts()
platforms

platforms
windows    27056
mac         8064
linux       5233
Name: count, dtype: int64

As we can see , most of the games are available on Windows platform , followed by Mac and Linux.